src/utils/errors/errorFormatter.js

```diff
@@ -13,14 +13,31 @@ export default function formatError (error) {
     return {
       message: 'Network error. Please check your internet connection.',
       code: 'NETWORK_ERROR',
-      details: {}
+      details: {
+        originalErrorMessage: error.message,
+        originalStack: error.stack
+      }
+    }
+  }
+
+  if (error instanceof TypeError && error.message.includes('Failed to parse URL')) {
+    return {
+      message: 'URL defined to fetch failed.',
+      code: 'URL_NOT_VALID',
+      details: {
+        originalErrorMessage: error.message,
+        originalStack: error.stack
+      }
```

Comment:
Suggestion: I would suggest not nesting things to keep that object simple. As an alternative, we could either put `error.message` in `details`, and just add another entry for `stack`, such as:
```suggestion
      details: error.message,
      stack: error.stack
```
What do you think ?